generator/src/main/java/com/scottlogic/deg/generator/guice/DecisionTreeOptimiserProvider.java

```diff
@@ -16,26 +16,16 @@
 
 package com.scottlogic.deg.generator.guice;
 
-import com.google.inject.Inject;
 import com.google.inject.Provider;
 import com.scottlogic.deg.generator.decisiontree.DecisionTreeOptimiser;
 import com.scottlogic.deg.generator.decisiontree.MostProlificConstraintOptimiser;
-import com.scottlogic.deg.generator.decisiontree.NoopDecisionTreeOptimiser;
-import com.scottlogic.deg.generator.generation.GenerationConfigSource;
 
 public class DecisionTreeOptimiserProvider implements Provider<DecisionTreeOptimiser> {
```

Comment:
should longer have these providers as there is only one implementation.

they should use bind().to() instead of bind().toProvider.

alternatively, if you want to remove the interface for optimiser and partitioner, they will automatically bind, and not need the bind().to() either

Reply:
i was hoping someone was going to suggest something like that as i agree that having a provider when only have one implementation using is pointless.  I'll also delete the noop implemntations of the 2 interfaces too.